packages/blade/src/components/Popover/__tests__/Popover.web.test.tsx

```diff
@@ -0,0 +1,281 @@
+
+    const ControlledExample = () => {
+      const [isOpen, setIsOpen] = React.useState(true);
+
+      return (
+        <>
+          <Popover
+            content="Hello world"
+            isOpen={isOpen}
+            onOpenChange={({ isOpen }) => setIsOpen(isOpen)}
+          >
+            <Button onClick={() => setIsOpen(true)}>{buttonText}</Button>
+          </Popover>
+          <Button onClick={() => setIsOpen((prev) => !prev)}>{toggleButtonText}</Button>
+        </>
+      );
+    };
+    const { getByRole, queryByRole } = renderWithTheme(<ControlledExample />);
+    await waitForPosition();
+
+    // should be opened by default
+    expect(queryByRole('dialog')).toBeInTheDocument();
+
+    // should close after 300ms
+    fireEvent.click(getByRole('button', { hidden: true, name: toggleButtonText }));
+
+    await act(async () => {
+      jest.advanceTimersByTime(299 + animationDuration);
+    });
+    expect(queryByRole('dialog')).not.toBeInTheDocument();
+
+    // open again
+    fireEvent.click(getByRole('button', { name: buttonText }));
+
+    await act(async () => {
+      jest.advanceTimersByTime(299);
+    });
+
+    expect(queryByRole('dialog')).toBeInTheDocument();
+
+    // close again with close icon
+    fireEvent.click(getByRole('button', { hidden: true, name: 'Close' }));
+
+    await act(async () => {
+      jest.advanceTimersByTime(299 + animationDuration);
+    });
+    expect(queryByRole('dialog')).not.toBeInTheDocument();
+  });
+
+  it('should have work with PopoverInteractiveWrapper', async () => {
+    jest.useFakeTimers();
+    const { getByTestId, queryByRole } = renderWithTheme(
+      <Popover content="Hello world">
+        <PopoverInteractiveWrapper>
+          <InfoIcon color="surface.action.icon.default.highContrast" size="medium" />
+        </PopoverInteractiveWrapper>
+      </Popover>,
+    );
+    const wrapper = getByTestId('popover-interactive-wrapper');
+    expect(wrapper).toHaveAttribute(
+      'data-blade-component',
+      MetaConstants.PopoverInteractiveWrapper,
+    );
+
+    // should open after 300ms
+    fireEvent.click(wrapper);
+    await act(async () => {
+      jest.advanceTimersByTime(300);
+    });
+    expect(queryByRole('dialog')).toHaveAttribute('data-blade-component', MetaConstants.Popover);
+
+    fireEvent.click(wrapper);
+
+    await act(async () => {
+      jest.advanceTimersByTime(299 + animationDuration);
+    });
+    expect(queryByRole('dialog')).not.toBeInTheDocument();
+  });
+
+  it('should pass a11y', async () => {
+    jest.useRealTimers();
+
+    const content = 'Hello world';
+    const title = 'Popover title';
+    const buttonText = 'Click me';
+    const { getByRole } = renderWithTheme(
+      <Popover defaultIsOpen title={title} content={content}>
+        <Button>{buttonText}</Button>
+      </Popover>,
+    );
+    expect(getByRole('dialog')).toHaveAccessibleName(title);
+    await assertAccessible(getByRole('dialog'));
+  });
```

Comment:
Another testcase - `should accept testID`

Reply:
Popover itself doesn't need the testID (even we aren't using it), it should be targeted by getByRole() or by the content inside of it. 

As for PopoverInteractiveWrapper, we have covered that case.